stores/migrations.go

```diff
@@ -645,6 +665,71 @@ func performMigration00013_uploadPackingOptimisations(txn *gorm.DB, logger *zap.
 			return err
 		}
 	}
-	logger.Info(context.Background(), "migration 00013_uploadPackingOptimisations complete")
+	logger.Info("migration 00013_uploadPackingOptimisations complete")
+	return nil
+}
+
+func performMigration00014_buckets(txn *gorm.DB, logger *zap.SugaredLogger) error {
+	logger.Info("performing migration 00014_buckets")
+	// Disable foreign keys in SQLite to avoid issues with updating constraints.
+	if isSQLite(txn) {
+		if err := txn.Exec(`PRAGMA foreign_keys = 0`).Error; err != nil {
+			return err
+		}
+	}
+
+	// Create buckets table
+	if !txn.Migrator().HasTable(&dbBucket{}) {
+		if err := txn.Migrator().CreateTable(&dbBucket{}); err != nil {
+			return err
+		}
+	}
+
+	// Add default bucket.
+	bucket := &dbBucket{
+		Name: api.DefaultBucketName,
+	}
+	if err := txn.FirstOrCreate(&bucket).Error; err != nil {
+		return err
+	}
+
+	// Add bucket id column to objects table.
+	if !txn.Migrator().HasColumn(&dbObject{}, "db_bucket_id") {
+		if err := txn.Migrator().AddColumn(&dbObject{}, "db_bucket_id"); err != nil {
+			return err
+		}
+	}
+
+	// Create missing composite index.
+	if !txn.Migrator().HasIndex(&dbObject{}, "idx_object_bucket") {
```

Comment:
why use the index name and not the field name?

Reply:
Because there are multiple indices on the column